cli/containerdutils/hostpaths_test.go

```diff
@@ -0,0 +1,183 @@
+package containerdutils
+
+import (
+	"context"
+	"fmt"
+	"syscall"
+	"testing"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cio"
+
+	"gotest.tools/assert"
+)
+
+func TestVerifyHostDirExistsNewContainerFail(t *testing.T) {
+	ctx := context.Background()
+	image := &fakeImage{}
+	dirname := "dirnamegoeshere"
+	client := &fakeContainerdClient{
+		newContainerFunc: func(ctx context.Context, id string, opts ...containerd.NewContainerOpts) (containerd.Container, error) {
+			return nil, fmt.Errorf("new container failure")
+		},
+	}
+	err := VerifyHostDirExists(ctx, client, dirname, image)
+	assert.ErrorContains(t, err, "new container failure")
+}
+
+func TestVerifyHostDirExistsFirstTaskRandomFail(t *testing.T) {
+	ctx := context.Background()
+	image := &fakeImage{}
+	dirname := "/dir/namegoeshere"
+	container := &fakeContainer{
+		newTaskFunc: func(context.Context, cio.Creator, ...containerd.NewTaskOpts) (containerd.Task, error) {
+			return nil, fmt.Errorf("task create failure")
+		},
+	}
+	client := &fakeContainerdClient{
+		newContainerFunc: func(ctx context.Context, id string, opts ...containerd.NewContainerOpts) (containerd.Container, error) {
+
+			return container, nil
+		},
+	}
+	err := VerifyHostDirExists(ctx, client, dirname, image)
+	assert.ErrorContains(t, err, "Failed to evaluate")
+}
+
+func TestVerifyHostDirExistsFirstTaskKillfail(t *testing.T) {
+	ctx := context.Background()
+	image := &fakeImage{}
+	dirname := "/dir/namegoeshere"
+	task := &fakeTask{
+		killFunc: func(context.Context, syscall.Signal, ...containerd.KillOpts) error {
+			return fmt.Errorf("task kill error")
+
+		},
+	}
+	container := &fakeContainer{
+		newTaskFunc: func(context.Context, cio.Creator, ...containerd.NewTaskOpts) (containerd.Task, error) {
+			return task, nil
+		},
+	}
+	client := &fakeContainerdClient{
+		newContainerFunc: func(ctx context.Context, id string, opts ...containerd.NewContainerOpts) (containerd.Container, error) {
+
+			return container, nil
+		},
+	}
+	err := VerifyHostDirExists(ctx, client, dirname, image)
+	assert.ErrorContains(t, err, "task kill error")
+}
+
+func TestVerifyHostDirExistsFirstTaskWaitfail(t *testing.T) {
+	ctx := context.Background()
+	image := &fakeImage{}
+	dirname := "/dir/namegoeshere"
+	task := &fakeTask{
+		waitFunc: func(context.Context) (<-chan containerd.ExitStatus, error) {
+			return nil, fmt.Errorf("wait failure")
+		},
+	}
+	container := &fakeContainer{
+		newTaskFunc: func(context.Context, cio.Creator, ...containerd.NewTaskOpts) (containerd.Task, error) {
+			return task, nil
+		},
+	}
+	client := &fakeContainerdClient{
+		newContainerFunc: func(ctx context.Context, id string, opts ...containerd.NewContainerOpts) (containerd.Container, error) {
+
+			return container, nil
+		},
+	}
+	err := VerifyHostDirExists(ctx, client, dirname, image)
+	assert.ErrorContains(t, err, "wait failure")
```

Comment:
Can we make sure to provide real typed errors for things instead of requiring checking error messages?